tests/nataliia_romanchuk/homework_4/test_collections.py

```diff
@@ -0,0 +1,424 @@
+    a, b, c, d, *tail = some_list
+    assert (a, b, c, d, tail) == ('a', 'b', 'c', 'd', [])
+
+
+def test_unpacking_sets_a_b_c_d_tail():
+    a, b, c, d, *tail = sorted(some_set)
+    assert a == 'a'
+    assert b == 'b'
+    assert c == 'c'
+    assert d == 'd'
+    assert set(tail) == set()
+
+
+def test_unpacking_dicts_a_b_c_d_tail():
+    a, b, c, d, *tail = sorted(some_dict)
+    assert a == 'a'
+    assert b == 'b'
+    assert c == 'c'
+    assert d == 'd'
+    assert set(tail) == set()
+
+
+# iterate protocol
+def test_iterator_for_string():
+    some_str_iter = iter(some_str)
+    first_item = next(some_str_iter)
+    assert first_item == 'a'
+
+    rest_items = list(some_str_iter)
+    assert rest_items == ['b', 'c', 'd']
+
+    exhausted_items = list(some_str_iter)
+    assert exhausted_items == []
+
+
+def test_iterator_for_tuple():
+    some_tuple_iter = iter(some_tuple)
+
+    first_item = next(some_tuple_iter)
+    assert first_item == 'a'
+
+    rest_items = list(some_tuple_iter)
+    assert rest_items == ['b', 'c', 'd']
+
+    exhausted_items = list(some_tuple_iter)
+    assert exhausted_items == []
+
+
+def test_iterator_for_list():
+    some_list_iter = iter(some_list)
+
+    first_item = next(some_list_iter)
+    assert first_item == 'a'
+
+    rest_items = list(some_list_iter)
+    assert rest_items == ['b', 'c', 'd']
+
+    exhausted_items = list(some_list_iter)
+    assert exhausted_items == []
+
+
+def test_iterator_for_set():
+    sort = sorted(some_set)
+    some_set_iter = iter(sort)
+
+    first_item = next(some_set_iter)
+    assert first_item in {'a', 'b', 'c', 'd'}
+
+    rest_items = list(some_set_iter)
+    assert set(rest_items) == {'b', 'c', 'd'}
+
+    exhausted_items = list(some_set_iter)
+    assert exhausted_items == []
+
+
+def test_iterator_for_dict():
+    some_dict_iter = iter(some_dict)
+
+    first_key = next(some_dict_iter)
+    assert first_key in {'a', 'b', 'c', 'd'}
+
+    rest_keys = list(some_dict_iter)
+    assert set(rest_keys) == {'b', 'c', 'd'}
+
+    exhausted_keys = list(some_dict_iter)
+    assert exhausted_keys == []
+
+
+# sort data (same for min max)
+data = [{"age": 16, "name": "John", "sex": "M"},
+        {"age": 34, "name": "Marry", "sex": "F"},
+        {"age": 25, "name": "Mathew", "sex": "M"}]
+
+
+def test_sort_data_by_age():
+    sorted_data_ascending = sorted(data, key=lambda i: i["age"])
+    sorted_data_descending = sorted(data, key=lambda i: i["age"], reverse=True)
+    assert sorted_data_ascending == [{"age": 16, "name": "John", "sex": "M"},
+                                     {"age": 25, "name": "Mathew", "sex": "M"},
+                                     {"age": 34, "name": "Marry", "sex": "F"}]
+
+    assert sorted_data_descending == [{"age": 34, "name": "Marry", "sex": "F"},
+                                      {"age": 25, "name": "Mathew", "sex": "M"},
+                                      {"age": 16, "name": "John", "sex": "M"}]
```

Comment:
assert [i["age"] for i in sorted_data_assending] == [16, 25, 34]
assert [i["age"] for i in sorted_data_descending] == [34, 25, 16]

Reply:
fixed